pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1221,6 +1229,7 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
 
 	// To make sure, that we set the right qemu wrapper arguments,
 	// we update the domain XML whenever a VirtualMachineInstance was already defined but not running
+	// FIXME: This does not work as we lose all the changes done to the domain spec in preStartHook
```

Comment:
Do we already do that, or do you talk about your changes? Let's not change the domain spec in the preStartHook. The domain-spec from the kubevirt-perspective should be final and only hooks, in the sense of external hooks, should modify it at this stage.

Reply:
This problem is already happening and is not related to my changes. Any changes done in the pre-start hook (for example, network bindings are resolved in the pre-start hook) will get lost if a defined non-running VM spec is re-synchronized afterwards. I have observed this many times on failure scenarios where the first startup will fail because something done on the pre-start hook causes problems, and on further attempts it will fail because the pre-start hook changes are not considered. Of course this is not observed most of the time because the normal state of a VM is either to be running or undefined on synchronizations.

Reply:
hm, that sounds like there are bugs in the retry logic then. Would be great to get a bug report next time you see it. Thanks for the details.

Reply:
@AlonaKaplan FYI, maybe we should look through our code parts there again and try to detect non-idempotent  steps.